src/main/java/org/gradle/playframework/tools/internal/routes/DefaultRoutesPostProcessor.java

```diff
@@ -0,0 +1,50 @@
+package org.gradle.playframework.tools.internal.routes;
+
+import org.gradle.util.RelativePathUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Serializable;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.stream.Stream;
+
+class DefaultRoutesPostProcessor implements Serializable {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DefaultRoutesPostProcessor.class);
+
+    void execute(RoutesCompileSpec spec) {
+        String sourceReplacementString = getSourceReplacementString(spec.getSources(), spec.getProjectDir());
+
+        try (Stream<Path> stream = Files.find(spec.getDestinationDir().toPath(), Integer.MAX_VALUE, (filePath, fileAttr) -> fileAttr.isRegularFile())) {
+            stream.forEach(routeFile -> process(routeFile, sourceReplacementString));
```

Comment:
the `process` function already catches the `IOException`. what else can throw it in this block?

Reply:
Files.find does